assignment-1/main.go

```diff
@@ -0,0 +1,56 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"time"
+
+	externalSnapshotClient "github.com/kubernetes-csi/external-snapshotter/client/v6/clientset/versioned"
+	snapshotClient "github.com/shahpratikr/k8s-dev-assignments/assignment-1/pkg/client/clientset/versioned"
+	snapshotInformer "github.com/shahpratikr/k8s-dev-assignments/assignment-1/pkg/client/informers/externalversions"
+	"github.com/shahpratikr/k8s-dev-assignments/assignment-1/pkg/controller"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+)
+
+func main() {
+	var (
+		config *rest.Config
+		err    error
+	)
+	localKubeconfigPath := homedir.HomeDir() + "/.kube/config"
+	kubeconfig := flag.String("kubeconfig", localKubeconfigPath, "location for kubeconfig file")
+	config, err = clientcmd.BuildConfigFromFlags("", *kubeconfig)
+	if err != nil {
+		config, err = rest.InClusterConfig()
+		if err != nil {
+			fmt.Println(err.Error())
+			return
+		}
+	}
+	snapshotClientSet, err := snapshotClient.NewForConfig(config)
+	if err != nil {
+		fmt.Println(err.Error())
+		return
+	}
+	kubernetesClientSet, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		fmt.Println(err.Error())
+		return
+	}
+	externalSnapshotClientSet, err := externalSnapshotClient.NewForConfig(config)
+	if err != nil {
+		fmt.Println(err.Error())
+		return
+	}
+	ch := make(chan struct{})
+	informerFactory := snapshotInformer.NewSharedInformerFactory(snapshotClientSet, 10*time.Minute)
+	c := controller.NewController(kubernetesClientSet, snapshotClientSet, externalSnapshotClientSet, informerFactory.Shahpratikr().V1alpha1().Snapshots())
+	informerFactory.Start(ch)
+	if err := c.Run(ch); err != nil {
+		fmt.Println(err.Error())
+		return
+	}
```

Comment:
We can improve main function. It should not have a lot of statements. Import a packages, initialize and run a method. All the client/informer creation can be moved to a helper function. You can also implement this methods on the `Controller` struct  e.g https://github.com/kanisterio/kanister/blob/master/cmd/controller/main.go